src/app/utils/hooks/useMeasure.ts

```diff
@@ -0,0 +1,36 @@
+import { MutableRefObject, useEffect, useState } from "react"
+
+type Position = {
+  x: number
+  y: number
+  width: number
+  height: number
+  pageX: number
+  pageY: number
+}
+
+export const useMeasure = ({ ref }: { ref: MutableRefObject<any> }) => {
+  const [position, setPosition] = useState<null | Position>(null)
+
+  useEffect(() => {
+    if (ref.current) {
+      ;(ref.current as any).measure(
+        (x: number, y: number, width: number, height: number, pageX: number, pageY: number) => {
+          setPosition({
+            x,
+            y,
+            width,
+            height,
+            pageX,
+            pageY,
+          })
+        },
+        () => {
+          console.error("measurement failed")
+        }
+      )
+    }
+  }, [JSON.stringify(position)])
```

Comment:
Shouldn't `position` work here too? When we change the position, we create a new object with `setPosition` anyway.

```suggestion
  }, [position])
```

Reply:
that's a good comment - i should have probably explained further why I am tricking this effect. 
The main reason behind me serializing the json is to make sure we're applying a shallow comparison the position object here. In the video attached, you can see the output of the console.log from both approaches.



Uploading Screen Recording 2023-04-28 at 12.52.34.mov…

Reply:
Thanks for the explanation 👍

Reply:
you're welcome of course 
sorry - the upload didn't work 

https://user-images.githubusercontent.com/11945712/235133362-a5c0b13b-f9c2-4e32-9af3-0c18d1ae2dfa.mov

Reply:
Isn't there a risk to create an infinite loop when setting position inside of the effect?

Reply:
yes, it's why I am making sure to compare the objects as a string. 
A better approach here would have been using a reference - issue is that the initial position always comes as 0. I can try it out